src/common/fetch/lens-fetch.injectable.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { Agent } from "https";
+import type { RequestInit, Response } from "node-fetch";
+import lensProxyPortInjectable from "../../main/lens-proxy/lens-proxy-port.injectable";
+import lensProxyCertificateInjectable from "../certificate/lens-proxy-certificate.injectable";
+import nodeFetchModuleInjectable from "./fetch-module.injectable";
+
+export type LensFetch = (pathnameAndOrQuery: string, init?: Omit<RequestInit, "agent">) => Promise<Response>;
+
+/**
+ * This fetch is for requesting items from the Lens Proxy and thus does not cause side effects
+ */
+const lensFetchInjectable = getInjectable({
```

Comment:
How about we override this in behavioural test with a fake that does whatever the "server" does. So basically maps the request to correct route in backend.

Reply:
Yeah sure.